src/Components/Endpoints/src/Builder/RazorComponentsEndpointRouteBuilderExtensions.cs

```diff
@@ -26,10 +29,42 @@ public static RazorComponentEndpointConventionBuilder MapRazorComponents<TRootCo
         ArgumentNullException.ThrowIfNull(endpoints);
 
         EnsureRazorComponentServices(endpoints);
+        AddBlazorWebJsEndpoint(endpoints);
 
         return GetOrCreateDataSource<TRootComponent>(endpoints).DefaultBuilder;
     }
 
+    private static void AddBlazorWebJsEndpoint(IEndpointRouteBuilder endpoints)
```

Comment:
We should add these endpoints to the data source, but ideally, we just make these files static web assets, and they get placed on your project. (not asking to do anything, just pointing it out).